src/networking.c

```diff
@@ -234,6 +266,9 @@ client *createClient(connection *conn) {
     c->commands_processed = 0;
     c->io_last_reply_block = NULL;
     c->io_last_bufpos = 0;
+    c->io_last_written_buf = NULL;
```

Comment:
Can't we instead use a bit flag in the payload header to indicate if we are done? Since the main thread needs to iterate over the headers inside the buffer anyway to get the actual written bytes.

Reply:
"the main thread needs to iterate over the headers inside the buffer anyway to get the actual written bytes" is not accurate. The main thread iterates over the headers inside the buffer **ONLY ONCE** when buffer should be released. 

"use a bit flag in the payload header" approach will cause:

1. main thread to iterate over the headers inside the buffer  even when buffer can't be released fully yet
2. writer thread to iterate over the headers inside the buffer in the end of write to detect/mark headers
 
The implemented `io_last_written_buf/io_last_written_bufpos/io_last_written_data_len ` eliminates redundant iterations over headers inside buffers and much simpler.